Add transient fault handling for loading passengers

ArthurAtangana opened the pull request:
## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] ✅ Test
- [ ] 🧑‍💻 Refactor

## Description

Add transient fault handling in the elevator: 1 second delay during load time when a passenger with a Transient fault is detected.

## How to test

Run the simulator using scenario 5 and/or 6.

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: already contain transient and combined fault test
- [ ] I need help with writing tests:.

## Screenshots
![Screenshot 2024-03-26 180412](https://github.com/ArthurAtangana/SYSC3303A_Project/assets/83664112/fadb19ba-497b-448d-9c19-ef75355ca7a2)

mpdesantis committed:
Add hard, transient, and combined fault test input files.

mpdesantis committed:
Update input files with fault column.

mpdesantis committed:
Add fault ingestion to Parser.

mpdesantis committed:
Update system config JSON to point to input file with all types of faults.

mpdesantis committed:
Update Scheduler's fault recognition for stopping elevators.

mpdesantis committed:
Condition prionts to help with debug.

WavyAM committed:
Added fault static method to generate Faults from int.

- Replaced parser switch with it (since it was broken)

WavyAM committed:
Adding hashCode override to pair with equals override:

- Hash structures use chain collisions to check for equality, not overriding it loses the collisions and does not allow us to even check for equality when there are no collisions.

WavyAM committed:
Reverting Scheduler fault matching to null as intended now that matching works.

braedenkloke committed:
feat(scheduler): add elevator timers and hard fault handling (#91)

* docs: add initial class diagram for timers

* feat(scheduler): stub start and kill timer methods

Additional add field to store elevator timers.

* feat(scheduler): add map of elevator timers

Timers are mapped to a given elevator number.

* docs: removed temporary timer class diagram

* feat: add timer task to handle fard fault.

Simply prints that a hard fault is detected ... then "takes the elevator out of service".

* fix: kill timer debugging logs only print when a timer is actually killed

* docs: cleanup javadocs and comments

Add fault handling for gophers.

* fix: add elevator timeout delay factor

1.5 times travel time.

mpdesantis committed:
Merge pull request #92 from ArthurAtangana/develop/parser-faults

Parser Fault Fields

mpdesantis reviewed (approved):
All these guys holding up elevators, man.

All looks good, and have verified it runs on my side!

ArthurAtangana committed:
Add transient fault handling for loading passengers

add 1 second delay to load time when a transient fault is detected on a passenger.

ArthurAtangana committed:
Add transient fault handling for loading passengers

add 1 second delay to load time when a transient fault is detected on a passenger.

ArthurAtangana committed:
Merge remote-tracking branch 'origin/develop/elevator-load-fault' into develop/elevator-load-fault
